### DISCUSSION TITLE
3.2.0-rc2

### DISCUSSION THREAD

**Discussion by sfmskywalker** [elsa-workflows/elsa-core#5719] [Design & tech]
## What's Changed
* Fix MT consumer registration when DisableConsumers is set by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5578
* Extended check when to update dependent workflows by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5598
* Revise ActivityRegistry update logic by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5604
* Forced package upgrade due to vulnerability by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5605
* Add HTTP file download activity and supporting classes by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5608
* Use auto detect of mySQL version by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5621
* Implement automated cleanup for Azure Service Bus subscriptions by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5636
* Update workflow definition registry and retraction handling by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5650
* Lazy loading of subgraphs by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5643
* Fix subscription name matching in RemoveOrphanedSubscriptions by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5658
* Update endpoints for lazy loading by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5681
* Update serialization and conversion functions by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5682
* Add sink for persisting workflows execution log records by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5708
* Workflow refresh feature by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-core/pull/5713

**Full Changelog**: https://github.com/elsa-workflows/elsa-core/compare/3.2.0-rc1...3.2.0-rc2

<hr /><em>This discussion was created from the release <a href='https://github.com/elsa-workflows/elsa-core/releases/tag/3.2.0-rc2'>3.2.0-rc2</a>.</em>

**Comment by glime-ai** [comment]
The changes between Elsa 3.2.0-rc1 and 3.2.0-rc2 include various fixes, updates, and new features. Here's a summary of the key changes:

1. **Fix MT consumer registration when DisableConsumers is set**: This pull request by @sfmskywalker addresses an issue where message transport (MT) consumers were not being registered correctly when `DisableConsumers` was set to `true`. The fix ensures that certain consumers, especially those involved with synchronizing local caches across the cluster, are still registered. [PR #5578](https://github.com/elsa-workflows/elsa-core/pull/5578)

2. **Extended check when to update dependent workflows**: This update by @raymonddenhaan improves the logic for determining when to update dependent workflows. [PR #5598](https://github.com/elsa-workflows/elsa-core/pull/5598)

3. **Revise ActivityRegistry update logic**: Another contribution by @raymonddenhaan revises the update logic for the ActivityRegistry. [PR #5604](https://github.com/elsa-workflows/elsa-core/pull/5604)

4. **Forced package upgrade due to vulnerability**: This pull request by @raymonddenhaan forces a package upgrade to address a security vulnerability. [PR #5605](https://github.com/elsa-workflows/elsa-core/pull/5605)

5. **Add HTTP file download activity and supporting classes**: @sfmskywalker added a new activity for HTTP file downloads along with supporting classes. [PR #5608](https://github.com/elsa-workflows/elsa-core/pull/5608)

6. **Use auto detect of MySQL version**: This change by @raymonddenhaan implements automatic detection of the MySQL version. [PR #5621](https://github.com/elsa-workflows/elsa-core/pull/5621)

7. **Implement automated cleanup for Azure Service Bus subscriptions**: This pull request by @raymonddenhaan adds automated cleanup for orphaned Azure Service Bus subscriptions. [PR #5636](https://github.com/elsa-workflows/elsa-core/pull/5636)

8. **Update workflow definition registry and retraction handling**: This update by @raymonddenhaan improves the workflow definition registry and the handling of workflow retractions. [PR #5650](https://github.com/elsa-workflows/elsa-core/pull/5650)

9. **Lazy loading of subgraphs**: @sfmskywalker implemented lazy loading for subgraphs to improve performance. [PR #5643](https://github.com/elsa-workflows/elsa-core/pull/5643)

10. **Fix subscription name matching in RemoveOrphanedSubscriptions**: This fix by @raymonddenhaan corrects the matching of subscription names in the RemoveOrphanedSubscriptions feature. [PR #5658](https://github.com/elsa-workflows/elsa-core/pull/5658)

11. **Update endpoints for lazy loading**: @sfmskywalker made updates to endpoints to support lazy loading. [PR #5681](https://github.com/elsa-workflows/elsa-core/pull/5681)

12. **Update serialization and conversion functions**: This update by @raymonddenhaan revises serialization and conversion functions. [PR #5682](https://github.com/elsa-workflows/elsa-core/pull/5682)

13. **Add sink for persisting workflows execution log records**: This new feature by @raymonddenhaan adds a sink for persisting workflow execution logs. [PR #5708](https://github.com/elsa-workflows/elsa-core/pull/5708)

14. **Workflow refresh feature**: @MariusVuscanNx contributed a feature for refreshing workflows. [PR #5713](https://github.com/elsa-workflows/elsa-core/pull/5713)

For more details, you can view the full changelog on the GitHub release page: [3.2.0-rc2 Changelog](https://github.com/elsa-workflows/elsa-core/compare/3.2.0-rc1...3.2.0-rc2).

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
